test/tests/functional/pbs_maintenance_reservations.py

```diff
@@ -0,0 +1,192 @@
+
+        msg = ""
+
+        try:
+            self.server.submit(r)
+        except PbsSubmitError as err:
+            msg = err.msg[0].strip()
+
+        self.assertTrue("pbs_rsub: can't use -l select with --hosts" == msg)
+
+        a = {'Resource_List.place': 'scatter',
+             '--hosts': self.mom.shortname}
+        r = Reservation(TEST_USER, attrs=a)
+
+        msg = ""
+
+        try:
+            self.server.submit(r)
+        except PbsSubmitError as err:
+            msg = err.msg[0].strip()
+
+        self.assertTrue("pbs_rsub: can't use -l place with --hosts" == msg)
+
+    def test_maintenance_confirm(self):
+        """
+        Test if the maintenance (prefixed with 'M') is immediately
+        confirmed without scheduler and the select, place, and resv_nodes
+        are correctly crafted. Also check if resv_enable = False is
+        ignored.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': False})
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'managers': '%s@*' % TEST_USER})
+
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes('vn', a, num=2, mom=self.mom)
+
+        a = {'resv_enable': False}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+        self.server.manager(MGR_CMD_SET, NODE, a, 'vn[0]')
+        self.server.manager(MGR_CMD_SET, NODE, a, 'vn[1]')
+
+        a = {'--hosts': self.mom.shortname}
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+
+        self.assertTrue(rid.startswith('M'))
+
+        exp_attr = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2'),
+                    'Resource_List.select':
+                    'host=%s:ncpus=4' % self.mom.shortname,
+                    'Resource_List.place': 'exclhost',
+                    'resv_nodes': '(vn[0]:ncpus=2)+(vn[1]:ncpus=2)'}
+        self.server.expect(RESV, exp_attr, id=rid)
+
+    def test_maintenance_degrade_reservation(self):
+        """
+        Test if the reservation is degraded by overlapping
+        maintenance reservation.
+        """
+        now = int(time.time())
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'managers': '%s@*' % TEST_USER})
+
+        a1 = {'Resource_List.select': 'host=%s' % self.mom.shortname,
+              'reserve_start': now + 3600,
+              'reserve_end': now + 3605}
+        r1 = Reservation(TEST_USER, attrs=a1)
+        rid1 = self.server.submit(r1)
+
+        exp_attr = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, exp_attr, id=rid1)
+
+        a2 = {'reserve_start': now + 3602,
+              'reserve_end': now + 3604,
+              '--hosts': self.mom.shortname}
+        r2 = Reservation(TEST_USER, attrs=a2)
+        self.server.submit(r2)
+
+        exp_attr = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10'),
+                    'reserve_substate': 12}
```

Comment:
I thought maintenance reservations marked overlapping reservations in conflict instead of degraded.  Also check for the shortened resv_nodes

Reply:
The state is DEGRADED. Only the substate is IN_CONFLICT, which is the '12'. It is possible to set also the state to IN_CONFLICT (just for being more obvious), but it is not necessary for the feature. Should I set also the state to IN_CONFLICT?

Reply:
Oh, I thought degraded was a substate.  The reservation was in state confirmed or running, but the substate was degraded.  This would degrade the reservation.  I guess I'm not as familiar with the degraded reservation code as I thought.